### PR TITLE
feat(auth): add google oauth into Airflow

### PR DESCRIPTION
**Pull request by david30907d**
## Types of changes

- **New feature**

## Description

add google OAuth
## Checklist:
- [x] Add test cases to all the changes you introduce
- [x] Run `poetry run pytest` locally to ensure all linter checks pass
- [ ] Update the documentation if necessary

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,9 +39,11 @@ Using Airflow to implement our ETL pipelines
 1. Build docker image: `docker build -t davidtnfsh/pycon_etl:cache --cache-from davidtnfsh/pycon_etl:cache .`
 2. Start the Airflow server: `docker run  --rm -p 80:8080 --name airflow  -v $(pwd)/dags:/usr/local/airflow/dags -v $(pwd)/service-account.json:/usr/local/airflow/service-account.json davidtnfsh/pycon_etl:cache webserver`
     * service-account.json: Please contact @david30907d using email, telegram or discord.
-3. Setup the Authentication of GCP: <https://googleapis.dev/python/google-api-core/latest/auth.html>
+3. Fill out `airflow.cfg` with Google OAuth ID and credential (ref: <https://airflow.apache.org/docs/apache-airflow/1.10.1/security.html#setting-up-google-authentication>)
```
Comment (Lee-W) [inline README.md:42]: I'd suggest against using bare URL. `Ref: [setting-up-google-authentication](https://airflow.apache.org/docs/apache-airflow/1.10.1/security.html#setting-up-google-authentication)` might be a better option.

File: airflow.cfg [hunk 2]
```diff
@@ -0,0 +1,997 @@
...
+# Whether to load the examples that ship with Airflow. It's good to
+# get started, but you probably want to set this to False in a production
+# environment
+load_examples = True
+
+# Where your Airflow plugins are stored
+plugins_folder = /usr/local/airflow/plugins
+
+# Secret key to save connection passwords in the db
+fernet_key = $FERNET_KEY
```
Comment (Lee-W) [inline airflow.cfg:136]: This looks like an env var to me. Should we add it to the setup step in README?

File: airflow.cfg [hunk 3]
```diff
@@ -0,0 +1,997 @@
...
+# to acknowledge the task before the message is redelivered to another worker.
+# Make sure to increase the visibility timeout to match the time of the longest
+# ETA you're planning to use.
+# visibility_timeout is only supported for Redis and SQS celery brokers.
+# See:
+# http://docs.celeryproject.org/en/master/userguide/configuration.html#std:setting-broker_transport_options
+# Example: visibility_timeout = 21600
+# visibility_timeout =
+
+[dask]
```
Comment (Lee-W) [inline airflow.cfg:529]: Are we using dask?

File: pyproject.toml [hunk 4]
```diff
@@ -7,19 +7,20 @@ authors = ["davidtnfsh <davidtnfsh@gmail.com>"]
...
 
 [tool.poetry.dev-dependencies]
 safety = "^1.9.0"
 bandit = "^1.6.2"
+pytest-cov = "^2.8.1"
+black = "^19.10b0"
+mypy = "^0.770"
+pytest = "^5.4.1"
+flake8 = "^3.8.3"
+diagrams = "^0.17.0"
```
Comment (Lee-W) [inline pyproject.toml:22]: What is `diagrams` for?

File: Dockerfile.test [hunk 5]
```diff
@@ -1,3 +1,4 @@
 FROM davidtnfsh/pycon_etl:cache
 ENV AIRFLOW_TEST_MODE=True
-USER root
\ No newline at end of file
+USER root
+COPY fixtures/airflow.test.cfg airflow.cfg
```
Comment (Lee-W) [inline Dockerfile.test:4]: Why do we put it in `fixtures`? Is it an airflow convention?

File: airflow.cfg (hunk 2 above)
Comment (david30907d) [reply airflow.cfg:136]: This env would be generated by entrypoint.sh automatically so might be ok him haha

here's the [ref](https://github.com/puckel/docker-airflow/blob/bed777970caa3e555ef618d84be07404438c27e3/script/entrypoint.sh#L12)

File: README.md (hunk 1 above)
Comment (david30907d) [reply README.md:42]: make sense to me~

File: airflow.cfg (hunk 3 above)
Comment (david30907d) [reply airflow.cfg:529]: no idea, i just copy the `airflow.cfg` template from this [repo](https://github.com/puckel/docker-airflow/blob/master/config/airflow.cfg)

any better cfg template would be welcome haha~

File: Dockerfile.test (hunk 5 above)
Comment (david30907d) [reply Dockerfile.test:4]: no it's just my convention 😅 

any suggestion to put these kinds of testing config?

File: pyproject.toml (hunk 4 above)
Comment (david30907d) [reply pyproject.toml]: long long time ago @tai271828 had recommended this tool at [discord](https://discord.com/channels/752904426057892052/755969596162441266/763802433817411624)

I integrated it into Airflow but now has been deprecated 😅 

will remove this one

File: Dockerfile.test (hunk 5 above)
Comment (Lee-W) [reply Dockerfile.test:4]: I would simply put this file in the root directory. Just curious. Why use the word `fixture`?

File: airflow.cfg (hunk 2 above)
Comment (Lee-W) [reply airflow.cfg:136]: Got it. Thanks for the clarification. 💯

File: airflow.cfg (hunk 3 above)
Comment (Lee-W) [reply airflow.cfg:529]: Ok, if we're to remove anything, we could do that in another PR.

Comment (david30907d) [reply airflow.cfg:529]: Sounds great~

File: Dockerfile.test (hunk 5 above)
Comment (david30907d) [reply Dockerfile.test:4]: Sure, will put it in the root folder.
Just used to put every test-related file into fixtures folder, but okay with either

**Review by Lee-W — APPROVED** [review]
Thanks for your update! Let's merge it 🎉🎉🎉

**Pull request merged**